cromshell

```diff
@@ -1471,6 +1488,99 @@ function fetch-all()
   return 0
 }
 
+# Get the cost for a workflow ID:
+function cost()
+{
+  which bq &>/dev/null   
+  local r=$?
+  [[ ${r} -ne 0 ]] && error "bq does not exist.  Must install the big query command-line client." && exit 8
+
+  [ ! -e ${BQ_COST_TABLE_FILE} ] && error "Big Query cost table file does not exist.  Must populate ${BQ_COST_TABLE_FILE} with big query cost table information." && exit 9
+
+  local cost_table=$(head -n1 ${BQ_COST_TABLE_FILE})
+  local id=$1
+
+  if [[ "$(uname)" == "Darwin" ]] ; then
+    local DATE_CMD=gdate
+  else
+    local DATE_CMD=date
+  fi  
+
+  local submissionDate=$( grep "${id}" ${CROMWELL_SUBMISSIONS_FILE} | head -n1 | awk '{print $1}' )
+  local y=$( echo ${submissionDate} | sed 's#^\([0-9][0-9][0-9][0-9]\).*#\1#' )
+  local m=$( echo ${submissionDate} | sed 's#^[0-9][0-9][0-9][0-9]\([0-9][0-9]\).*#\1#' )
+  local d=$( echo ${submissionDate} | sed 's#^[0-9][0-9][0-9][0-9][0-9][0-9]\([0-9][0-9]\).*#\1#' )
+
+  local START_DATE="${y}-${m}-${d}"
+  local END_DATE=$($DATE_CMD +%Y-%m-%d -d "${START_DATE} +4 days")
```

Comment:
Should we deal with the case where gdate is missing?

Reply:
Yeah.  I don't really want to code around having it, so I'll have cromshell throw an error.